### DISCUSSION TITLE
Widget in React

### DISCUSSION THREAD

**Discussion by hdzierz** [visgl/deck.gl#9046] [General]
I have a question: I would like to add a ZoomWidget to a React page. I was wondering whether someone coudl tell me how to do that as there does not seem to be a good description,

**Comment by chrisgervang** [comment]
You can use the widgets in react by supplying it in a `widgets` props on `<DeckGL widgets={[new ZoomWidget({})]}/>`.

We're working on a JSX syntax for widgets but they should work similarly, just a different syntax.

**Reply by borjaevo** [reply]
> You can use the widgets in react by supplying it in a `widgets` props on `<DeckGL widgets={[new ZoomWidget({})]}/>`.
> 
> We're working on a JSX syntax for widgets but they should work similarly, just a different syntax.

I tried to use it this way or the new way, adding a widget as a child, for example <ZoomWidget/>. But the zoomwidget is shown as two very small rectangular boxes that you can't click or use it. Also sometimes renders the widget mutiple times and the widget is shown 2, 3 or 5 times.

**Reply by unautre** [reply]
Hello,

> [...] Also sometimes renders the widget mutiple times and the widget is shown 2, 3 or 5 times.

I seem to end up with the same issue. Have you ever found a solution ?

**Reply by amarant** [reply]
known issue : https://github.com/visgl/deck.gl/issues/9957
